src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts

```diff
@@ -1 +1,132 @@
+import { AlterationTypeConstants } from '../../../pages/resultsView/ResultsViewPageStore';
+import client from 'shared/api/cbioportalClientInstance';
+import {
+    GenericAssayMetaFilter,
+    GenericAssayDataFilter,
+    GenericAssayData,
+    GenericAssayMeta,
+} from 'shared/api/generated/CBioPortalAPI';
+import { MolecularProfile } from 'shared/api/generated/CBioPortalAPI';
+import _ from 'lodash';
+import { IDataQueryFilter } from '../StoreUtils';
+import { GenericAssayDataMultipleStudyFilter } from 'shared/api/generated/CBioPortalAPI';
+
 export const NOT_APPLICABLE_VALUE = 'NA';
+
+export async function fetchGenericAssayMetaByMolecularProfileIdsGroupByGenericAssayType(
+    molecularProfiles: MolecularProfile[]
+) {
+    const genericAssayProfiles = molecularProfiles.filter(profile => {
+        return (
+            profile.molecularAlterationType ===
+            AlterationTypeConstants.GENERIC_ASSAY
+        );
+    });
+
+    const genericAssayProfilesGroupByGenericAssayType = _.groupBy(
+        genericAssayProfiles,
+        'genericAssayType'
+    );
+    const genericAssayTypes = _.keys(
+        genericAssayProfilesGroupByGenericAssayType
+    );
+
+    const GenericAssayMetaGroupByGenericAssayType: {
+        [genericAssayType: string]: GenericAssayMeta[];
+    } = {};
+
+    await Promise.all(
+        genericAssayTypes.map(async genericAssayType => {
+            const genericAssayProfiles =
+                genericAssayProfilesGroupByGenericAssayType[genericAssayType];
+            const genericAssayProfileIds = genericAssayProfiles.map(
+                profile => profile.molecularProfileId
+            );
+            GenericAssayMetaGroupByGenericAssayType[
+                genericAssayType
+            ] = await fetchGenericAssayMetaByProfileIds(genericAssayProfileIds);
+        })
+    );
+    return GenericAssayMetaGroupByGenericAssayType;
+}
+
+export async function fetchGenericAssayMetaByMolecularProfileIdsGroupByMolecularProfileId(
+    molecularProfiles: MolecularProfile[]
+) {
+    const genericAssayProfiles = molecularProfiles.filter(profile => {
+        return (
+            profile.molecularAlterationType ===
+            AlterationTypeConstants.GENERIC_ASSAY
+        );
+    });
+
+    const genericAssayMetaGroupByMolecularProfileId: {
+        [molecularProfileId: string]: GenericAssayMeta[];
+    } = {};
+
+    await Promise.all(
+        genericAssayProfiles.map(async genericAssayProfile => {
```

Comment:
use then and make sure the map callback actually returns a promise.

Reply:
disagree. async causes the function to return a promise so I think this makes sense

Reply:
@adamabeshouse i would argue that the only reason to use async await is to avoid using a callback.  it's syntatic sugar that allows us to pretend we are programming syncronously.  but we use a callback here anyway in order to populate genericAssayMetaGroupByMolecularProfileId, so i would say that the use of await here actually obscures what's happening.  it's same argument i make for using reduce  or map rather than forEach.    by using assignment, it makes the result of operation clear rather than forcing the developer to look at the internals of the loop to see that it's populating the empty thing declared above.

Reply:
I think I agree with @alisman, and I think that the clearest way to do it would be
```
await Promise.all(
    genericAssayProfiles.map(
        profile=>getData(profile).then(update genericAssayMetaGroupByMolecularProfileId with data)
    )
);
return genericAssayMetaGroupByMolecularProfileId;
```